src/utils/configuration.cpp

```diff
@@ -122,8 +123,15 @@ void ConfigurationMap::addKeys(keys_t& names) const
 bool ConfigurationEnviron::tryGetPropertyAsString(const std::string& name, std::string* val) const
 {
     const char *env = getenv(name.c_str());
-    if(!env || !*env)
+    if(!env || !*env) {
+		const ENV_PARAM *pParam;
+		pParam = envFindConfigParam( name.c_str() );
+		if ( pParam && pParam->pdflt ) {
+			*val = pParam->pdflt;
```

Comment:
My reason for passing `string* val` is to hint that it may be NULL.  cf. `Configuration::hasProperty()`.

Reply:
Yes, I should have included a test for NULL val ptr.   Hadn't seen hasProperty().